mule-module-apikit/src/main/java/org/mule/module/apikit/HttpRestRequest.java

```diff
@@ -143,6 +143,46 @@ public MuleEvent validate(IAction action) throws MuleException
         return requestEvent;
     }
 
+    private void processQueryString() throws InvalidQueryStringException
+    {
+        IParameter expected = action.getQueryString();
+        if (!shouldProcessQueryString(expected)) return;
+
+        Map queryParamsMap = requestEvent.getMessage().getInboundProperty("http.query.params");
+        String queryString = buildQueryString(expected, queryParamsMap);
+
+        if (!expected.validate(queryString))
+        {
+            throw new InvalidQueryStringException("Invalid value for query string ");
+        }
+    }
+
+    private boolean shouldProcessQueryString(IParameter queryString) {
+        return queryString != null && !queryString.isArray() && !queryString.isScalar();
+    }
+
+    private String buildQueryString(IParameter expected, Map queryParamsMap) {
+        StringBuilder result = new StringBuilder();
+
+        for (Object property : queryParamsMap.keySet()) {
+
+            Collection<?> actualQueryParam = getActualQueryParam(property.toString());
+            result.append("\n").append(property).append(": ");
+
+            if (actualQueryParam.size() > 1 || expected.isFacetArray(property.toString())) {
+                for (Object o : actualQueryParam) {
+                    result.append("\n  - ").append(o);
+                }
+            } else
```

Comment:
shouldn't we raise an error when 'actualQueryParam' has more than one value but the expected one is not an array?

Reply:
Validation won't pass in that case

Reply:
ok, this method only builds the queryString. Dismiss the comment, please